packages/components/src/internal/components/domainproperties/NameExpressionGenIdBanner.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import { NameExpressionGenIdBanner } from "./NameExpressionGenIdBanner";
+import {getTestAPIWrapper} from "../../APIWrapper";
+import {getDomainPropertiesTestAPIWrapper} from "./APIWrapper";
+import {sleep} from "../../testHelpers";
+
+describe('NameExpressionGenIdBanner', () => {
+
+    test('with existing data', async () => {
+        const tree = renderer.create(
+            <NameExpressionGenIdBanner
+                dataTypeName="Data1"
+                rowId={100}
+                kindName='DataClass'
+                api={getTestAPIWrapper(jest.fn, {
+                    domain: getDomainPropertiesTestAPIWrapper(jest.fn, {
+                        hasExistingDomainData: () => Promise.resolve(true),
+                        getGenId: () => Promise.resolve(123),
+                    }),
+                })}
+            />
+        );
+        await sleep();
+        await sleep();
+        expect(tree).toMatchSnapshot();
```

Comment:
Agree that snapshots are less ideal. But I was not able to get non snapshot tests done that uses await sleep().

Reply:
@XingY I would recommend using `await waitForLifecycle(tree);` instead of `sleep()`. This calls `act()` which is often necessary when testing components that use React Hooks.

Reply:
Cool. I did not know about that. Will close the PR as is and refactor in a future PR. thanks.